Editors/PathEditor.cpp

```diff
@@ -10,6 +10,19 @@
 #include <QSpinBox>
 #include <QToolButton>
 
+void initialize_static_path_header_data() {
+  auto descriptor = Path::Point::GetDescriptor();
+  auto x_name = descriptor->FindFieldByNumber(Path::Point::kXFieldNumber)->full_name();
+  auto y_name = descriptor->FindFieldByNumber(Path::Point::kYFieldNumber)->full_name();
+  auto speed_name = descriptor->FindFieldByNumber(Path::Point::kSpeedFieldNumber)->full_name();
+  ProtoModel::SetHeaderData(x_name, Qt::Horizontal, PathEditor::tr("X"), Qt::DisplayRole);
+  ProtoModel::SetHeaderData(y_name, Qt::Horizontal, PathEditor::tr("Y"), Qt::DisplayRole);
+  ProtoModel::SetHeaderData(speed_name, Qt::Horizontal, PathEditor::tr("Speed"), Qt::DisplayRole);
+  ProtoModel::SetHeaderData(x_name, Qt::Horizontal, QIcon(":/actions/diamond-red.png"), Qt::DecorationRole);
+  ProtoModel::SetHeaderData(y_name, Qt::Horizontal, QIcon(":/actions/diamond-green.png"), Qt::DecorationRole);
+  ProtoModel::SetHeaderData(speed_name, Qt::Horizontal, QIcon(":/actions/motion.png"), Qt::DecorationRole);
+}

```

Comment:
Here it is in all its glory, the static header data. Note I am using the static QObject translate, which I don't know if it effects dynamic translation of the UI at runtime (say for switching the language dynamically).